### PR TITLE
Add global flag to disable dev services

### LINKED ISSUE #19133: Global config property to disable all DevServices
### Description

It would be nice to have a global config property to be able to enable/disable all DevServices, something like `quarkus.devservices.enabled`.

A global switch to shut things off (like if I need/want to run `quarkus:dev` in a remote IDE like Eclipse Che which doesn't allow you to start containers because you're already in a container). In such a case if my app uses, say, a postgres db, kafka, & apicurio, it would be really nice to just be able to shut devservices off completely.

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #19133

### CODE REVIEW

File: extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java [hunk 1]
```diff
@@ -510,11 +510,13 @@ public Integer get() {
     }
 
     public ThreadFactory createThreadFactory() {
+        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
         AtomicInteger threadCount = new AtomicInteger(0);
         return runnable -> {
             VertxThread thread = VertxThreadFactory.INSTANCE.newVertxThread(runnable,
                     "executor-thread-" + threadCount.getAndIncrement(), true, 0, null);
             thread.setDaemon(true);
+            thread.setContextClassLoader(tccl);
```
Comment (gsmet) [inline extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java:519]: @stuartwdouglas this change doesn't look related? Is it something you want in?

**Review by gsmet — CHANGES_REQUESTED** [review]
Blocking until we clarify the TCCL change.

File: extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java (hunk 1 above)
Comment (stuartwdouglas) [reply extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java]: oops, fixed

**Review by machi1990 — COMMENTED** [review]
Nice addition. Will be good to add it in devservices documentation.

**Review by machi1990 — APPROVED** [review]
lgtm

**Review by gsmet — APPROVED** [review]
